index.js

```diff
@@ -61,15 +60,16 @@ var WinstonCloudWatch = function(options) {
 
 util.inherits(WinstonCloudWatch, winston.Transport);
 
-WinstonCloudWatch.prototype.log = function(level, msg, meta, callback) {
-  debug('log (called by winston)', level, msg, meta);
+WinstonCloudWatch.prototype.log = function (info, callback) {
+  var log = { level: info.level, msg: info.message };
```

Comment:
Using 3.x winston — the `msg` parameter is now a property on the `info` parameter, which is being handled properly here. The issue is the `meta` parameter. In 3.x, winston now attaches all other parameters passed to `logger[method]()` at the top level of the `info` object.

Here's an example:

```js
const somethingToLog = { hello: 'pat' }
logger.info('Time to log something', somethingToLog)
```

Using that example, the `info` parameter in `WinstonCloudWatch.prototype.log()` looks like the following:

```json
{
  "level": "info",
  "message": "Time to log something",
  "hello": "pat"
}
```

> The `info` parameter will also have `Symbol`s for `level` and `message` and any other settings set up by the *global* format (for example: using `format.timestamp()` will include a `timestamp` property on `info`).

The current implementation **only** makes the `message` and `level` available to the `WinstonCloudWatch.messageFormatter` function. I think that all of the information should be passed down to the formatter, so that it could also be logged.

I don't have a great solution, but an minimum I would think we could pass down the entire `info` parameter as a `meta` argument to the formatter and let the user decide what to do — `var log = { level: info.level, msg: info.message, meta: info };`.

This PR is great, and I'm using it in production today. Thanks for keeping it up to date!

Reply:
@patrickrgaffney we could just put it there in `meta` yes, if a particular need arises then we could talk about it when and if it happens.

Thank you for testing this in production, makes me particularly happy as it's a crucial test; I've left the company where I was using this in production so I don't have that kind of feedback.

Reply:
@lazywithclass I have an updated version of `WinstonCloudWatch.prototype.log` with those changes and supporting tests on [my fork](https://github.com/flybondi/winston-cloudwatch/compare/feature/winston3-support...patrickrgaffney:winston3) — I can push those change to this PR if you would like.

Reply:
@patrickrgaffney Very good news! And thanks to try it out.

BTW, forgot no install proper dependencies on CI job. Fixing the PR now.

Reply:
@patrickrgaffney I'm not particularly fond of passing duplicate data to the final log message (`level` and `message` would actually be repeated under `.meta`). What about my original suggestion of passing everything _other tan_ `level` or `message` as `meta`?

So for an `info` object such as `{ level: 'info', message: 'foo', hello: 'pat' }`, the Cloudwatch log would become `{ level: 'info, msg: 'foo', meta: { hello: 'pat' } }`. 

Or even better, let's use the raw `info` as the actual log. It there anything preventing us from doing that?

Reply:
I'm actually a bigger fan of the latter option — just exposing the raw `info` object. I don't see any reason why this wouldn't work — it's how the built-in *global* formatter works and it plays nicely with the object destructuring (if you're using node v8):

```js
const cloudWatch = new WinstonCloudWatch({
  messageFormatter: ({ level, message, ...meta }) => { /* ... */ },
})
```

I'll change my implementation to try this method and link it here later today so you can check it out. I wanted to add another test to check the params on the custom formatter anyway. 👍

Reply:
Okay, updated [my fork](https://github.com/flybondi/winston-cloudwatch/compare/feature/winston3-support...patrickrgaffney:winston3) to use pass raw `info` object to the `messageFormatter`. I also added another test — now we're testing both scenarios: logging JSON and logging text (with a custom formatter).

Check it out and let me know.

Reply:
just bumping this 🙂

Reply:
Sorry I had a look at it two weeks ago but forgot to reply, I agree with both of you about using `info` as is.

Reply:
Sounds good. I wrote an implementation in [my fork](https://github.com/flybondi/winston-cloudwatch/compare/feature/winston3-support...patrickrgaffney:winston3) that implements this fairly simply — i added a few tests as well.

Reply:
Yes I plan to use your work for sure :)